### PR TITLE
Also match for opensuse-welcome in first_boot/finish_desktop

### PR DESCRIPTION
**Pull request by StefanBruens**
Currently, needles with the openSUSE Welcome dialog are duplicated, once
with the "generic-desktop" tag and once with "opensuse-welcome". This
also makes it difficult to differentiate between a "true" generic desktop
and one showing the dialog, i.e. in the opensuse_welcome module.

While at it, remove the long defunct 'kde-greeter' handling.

- Verification run: https://openqa.opensuse.org/tests/1028495

### CODE REVIEW

File: tests/installation/finish_desktop.pm [hunk 1]
```diff
@@ -22,13 +22,7 @@ sub run {
 
     # live may take ages to boot
     my $timeout = 600;
-    assert_screen "generic-desktop", $timeout;
-
-    ## duplicated from second stage, combine!
-    if (check_var('DESKTOP', 'kde')) {
-        send_key "esc";
-        assert_screen "generic-desktop", 25;
-    }
```
Comment (okurz) [inline tests/installation/finish_desktop.pm]: removing this block seems like the right thing to do but still unrelated

File: tests/installation/first_boot.pm [hunk 2]
```diff
@@ -59,10 +59,7 @@ sub run {
         handle_login;
     }
 
-    my @tags = qw(generic-desktop);
-    if (check_var('DESKTOP', 'kde') && get_var('VERSION', '') =~ /^1[23]/) {
-        push(@tags, 'kde-greeter');
-    }
```
Comment (okurz) [inline tests/installation/first_boot.pm]: same here

Comment (StefanBruens) [reply tests/installation/first_boot.pm]: kde-greeter is the defunct, kde-specific precursor of opensuse-welcome. So IMHO it is not unrelated.

Comment (okurz) [reply tests/installation/first_boot.pm]: ok, fine

**Comment by DimStar77** [comment]
> you wanted to avoid the check for opensuse-welcome as well, right? WDYT?

Originally this was the plan, but seeing on how many problems we received so far, I'm perfectly willing to go the route Stefan proposes here. Especially as we later on need to be able to check 'generic desktop' where os-welcome is not present (to ensure it is closed)

File: tests/installation/finish_desktop.pm [hunk 3]
```diff
@@ -22,13 +22,7 @@ sub run {
...
+    assert_screen([qw(generic-desktop opensuse-welcome)], $timeout);
```
Comment (okurz) [inline tests/installation/finish_desktop.pm]: could you please only add the tag "opensuse-welcome" when we would expect the opensuse-welcome package to be installed? E.g. `is_opensuse && !is_leap('<15.2') && !is_live_cd` or something? If there isn't a `is_opensuse_welcome_applicable` maybe it's time to add it now :)

Comment (StefanBruens) [reply tests/installation/finish_desktop.pm]: there is `opensuse_welcome_applicable`, also see #8399

**Comment by StefanBruens** [comment]
Good to go?

**Pull request merged**